docs/src/domains/steps/non-production-custom-domains.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Set up a custom domain on your non-production environments
+sidebarTitle: Non-production custom domains
+weight: 2
+description: Learn how to set up custom domains on your staging and development environments
+betaFlag: true
+---
+
+{{< premium-features/tiered "Elite and Enterprise" >}}
+
+You can add a custom domain to a staging or development environment without modifying your [routes configuration](../../define-routes/_index.md).
+
+To do so, you need to attach the new non-production custom domain
+to an existing production custom domain.
+
+{{< note title="Example" >}}
+
+You have two environments, a production environment and a staging environment.
+You've added the `example.com` custom domain to your production environment.
+
+You can only create the `staging.example.com` custom domain if you attach it to the `example.com` custom domain. 
+
+You can then access your staging environment through `staging.example.com`
+and still access your production environment through `example.com`.
+
+{{< /note >}}
+
+If you have multiple production environments,
+when you create a non-production custom domain,
+Platform.sh retrieves the `{default}` placeholder in your [routes configuration](../../define-routes/_index.md).
+It then replaces it with the correct production custom domain
+according to what you specify [in the setup command](#add-a-custom-domain-to-a-non-production-environment).
```

Comment:
This is the hard part to explain, and after more thoughts, I think we could say something like this:

> If you have multiple custom domains on your production environment,
> when you create a non-production custom domain,
> you don't have to update your [routes configuration](../../define-routes/_index.md), 
> Platform.sh automatically figures out the routing of your non-production environment, based on your production custom domains in your `routes.yaml` file and the non-production custom domains attached to them. 

Let me know what you think.

Reply:
That's great. I only split the explanation into several sentences and a bulleted list for improved readability. Also removed the explicit mention of the .yaml file to remain consistent with our docs style guide.